packages/server/src/shared/internal/serialize.ts

```diff
@@ -36,12 +37,24 @@ type SerializeTuple<T extends [unknown, ...unknown[]]> = {
   [k in keyof T]: T[k] extends NonJsonPrimitive ? null : Serialize<T[k]>;
 };
 
+// prettier-ignore
+type SerializeObjectKey<T extends Record<any, any>, TKey> = 
+  // never include entries where the key is a symbol
+  TKey extends symbol ? never : 
+  // always include entries where the value is any
+  IsAny<T[TKey]> extends true ? TKey :
+  // always include entries where the value is unknown
+  unknown extends T[TKey] ? TKey : 
+  // never include entries where the value is a non-JSON primitive
+  T[TKey] extends NonJsonPrimitive ? never : 
+  // otherwise serialize the value
+  TKey;
```

Comment:
cc @jussisaurio i made this fn - looks good?

Reply:
yeah lgtm